libmcount/mcount.c

```diff
@@ -1237,12 +1237,20 @@ static int __mcount_entry(unsigned long *parent_loc, unsigned long child,
 	rstack->nr_events  = 0;
 	rstack->event_idx  = ARGBUF_SIZE;
 
-	/* hijack the return address of child */
-	*parent_loc = mcount_return_fn;
+	if (unlikely(mcount_flat)) {
```

Comment:
I don't think we need `unlikely` here since it'd hit 100% if the option is given.